camelot/view/controls/tableview.py

```diff
@@ -607,57 +614,66 @@ def set_filters(self, filters):
-        if filters:
+        if filter_routes:
             filters_widget = ActionsBox(parent=self)
             filters_widget.setObjectName('filters')
             self.filters_layout.addWidget(filters_widget)
-            for action in filters:
+            for action_route in filter_routes:
+                action = AdminRoute.action_for(tuple(action_route))
                 action_widget = self.render_action(action, filters_widget)
+                action_widget.current_row_changed_signal.connect(self.get_model().change_selection)
+                action_widget.set_state_v2(self._get_action_state(action_route, action_states))
                 filters_widget.layout().addWidget(action_widget)
         self.filters_layout.addStretch(1)
 
-    def set_list_actions(self, actions):
+    def set_list_actions(self, action_routes, action_states):
         """sets filters for the tableview"""
         assert object_thread(self)
         actions_widget = self.findChild(ActionsBox, 'actions')
-        if actions:
+        if action_routes:
             actions_widget = ActionsBox(parent=self)
             actions_widget.setObjectName('actions')
-            for action in actions:
-                actions_widget.layout().addWidget(
-                    self.render_action(action, actions_widget)
-                )
+            for action_route in action_routes:
+                action = AdminRoute.action_for(tuple(action_route))
+                action_widget = self.render_action(action, actions_widget)
+                action_widget.current_row_changed_signal.connect(self.get_model().change_selection)
+                action_widget.set_state_v2(self._get_action_state(action_route, action_states))
+                actions_widget.layout().addWidget(action_widget)
             self.filters_layout.addWidget(actions_widget)
 
     @QtCore.qt_slot( object, object )
-    def set_toolbar_actions( self, toolbar_area, toolbar_actions ):
+    def set_toolbar_actions( self, toolbar_area, action_routes, action_states ):
         """Set the toolbar for a specific area
         :param toolbar_area: the area on which to put the toolbar, from
             :class:`Qt.LeftToolBarArea` through :class:`Qt.BottomToolBarArea`
         :param toolbar_actions: a list of :class:`camelot.admin.action..base.Action` objects,
             as returned by the :meth:`camelot.admin.application_admin.ApplicationAdmin.get_toolbar_actions`
             method.
         """
-        if toolbar_actions != None:
+        if action_routes != None:
             toolbar = self.findChild(QtWidgets.QToolBar, 'actions_toolbar')
             assert toolbar
-            for action in toolbar_actions:
+            for action_route in action_routes:
+                action = AdminRoute.action_for(tuple(action_route))
                 rendered = self.render_action(action, toolbar)
+                rendered.current_row_changed_signal.connect(self.get_model().change_selection)
```

Comment:
ik denk niet dat die link tss het selection model and de collection proxy via de action widget moet gaan ?

kunnen we die 2 niet gewoon bij constructie van de view connecteren ?